src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -8389,6 +8416,7 @@ const QList<QgsExpressionFunction *> &QgsExpression::Functions()
         << new QgsStaticExpressionFunction( QStringLiteral( "array_to_string" ), QgsExpressionFunction::ParameterList() << QgsExpressionFunction::Parameter( QStringLiteral( "array" ) ) << QgsExpressionFunction::Parameter( QStringLiteral( "delimiter" ), true, "," ) << QgsExpressionFunction::Parameter( QStringLiteral( "emptyvalue" ), true, "" ), fcnArrayToString, QStringLiteral( "Arrays" ) )
         << new QgsStaticExpressionFunction( QStringLiteral( "string_to_array" ), QgsExpressionFunction::ParameterList() << QgsExpressionFunction::Parameter( QStringLiteral( "string" ) ) << QgsExpressionFunction::Parameter( QStringLiteral( "delimiter" ), true, "," ) << QgsExpressionFunction::Parameter( QStringLiteral( "emptyvalue" ), true, "" ), fcnStringToArray, QStringLiteral( "Arrays" ) )
         << new QgsStaticExpressionFunction( QStringLiteral( "generate_series" ), QgsExpressionFunction::ParameterList() << QgsExpressionFunction::Parameter( QStringLiteral( "start" ) ) << QgsExpressionFunction::Parameter( QStringLiteral( "stop" ) ) << QgsExpressionFunction::Parameter( QStringLiteral( "step" ), true, 1.0 ), fcnGenerateSeries, QStringLiteral( "Arrays" ) )
+        << new QgsStaticExpressionFunction( QStringLiteral( "geom_to_array" ), QgsExpressionFunction::ParameterList() << QgsExpressionFunction::Parameter( QStringLiteral( "geometries" ) ), fcnGeoCollAsArray, QStringLiteral( "Arrays" ) )
```

Comment:
@nirn I'd love a second opinion on the function name here -- do you think 'geom_to_array' is ok? I'm +0.

Reply:
Meant @nirvn ^

-----

Comment from the sideline:

As a lover of long_descriptive_names I am not a fan of the "geom" abbreviation already, so I am biased:

"geometrycollection_to_array" would be the longest function name to date (with 27 characters; "sqlite_fetch_and_increment" is 26, "longest_common_substring" is 24) but it would be descriptive and more intuitive. "geom_to_array" could be interpreted as something that creates an array of coordinates.

Expressions are probably usually edited in the expression dialog where there is plenty of space and no need for that kind of optimization. On the other hand, this IS quite long...

I'd vote for long, descriptive and not surprising but I don't know :)

Reply:
@kannes maybe `geometries_to_array` as it handle multiparts and the notion of 'geometrycollection' is not common in expressions.

Reply:
I like that! It would fit a thought like "take the (member) *geometries* of this GeometryCollection/Multi* and give them to me in an array."